tools/mason/mason.chpl

```diff
@@ -96,27 +96,40 @@ proc main(args: [] string) {
 
 
 proc masonClean() {
-  runCommand('rm -rf target');
+  try! {
+    const cwd = getEnv("PWD");
+    const projectHome = getTopLvlDirPath(cwd);
+    runCommand('rm -rf ' + projectHome + '/target');
+  }
+  catch e: MasonError {
+    writeln(e.message());
+  }
 }
 
 
 proc masonDoc(args) {
-  var toDoc = basename(getEnv('PWD'));
-  var project = toDoc + '.chpl';
-  if isDir('src/') {
-    if isFile('src/' + project) {
-      var command = 'chpldoc src/' + project;
-      writeln(command);
-      runCommand(command);
+  try! {
+    const cwd = getEnv("PWD");
+    const projectHome = getTopLvlDirPath(cwd);
+    const toDoc = basename(projectHome);
+    const project = toDoc + '.chpl';
+    if isDir(projectHome + '/src/') {
+      if isFile(projectHome + '/src/' + project) {
+        const command = 'chpldoc ' + projectHome + '/src/' + project + ' -o ' + projectHome + '/doc/';
```

Comment:
It would also be nice to test this functionality if that does not already exist

Reply:
this goes in line with something I was going to add to the epic now that we have better error handling which is a more sophisticated error class with better responses such as the one you just mentioned.